src/index_result.c

```diff
@@ -216,8 +218,7 @@ void IndexResult_Free(RSIndexResult *r) {
 
       AggregateResultIter_Free(iter);
     }
-    rm_free(r->data.agg.children);
-    r->data.agg.children = NULL;
+    AggregateResult_FreeChildren(&r->data.agg);
```

Comment:
This will just free the children. Where is the `Box<RSAggregateResult>` itself being freed? Don't we need a `AggregateResult_Free()`?

Reply:
Thanks for catching this!! Should be fixed in https://github.com/RediSearch/RediSearch/pull/6494/commits/a40f02900e871858bd602ebb9bc25b7af6e88857